docs/conf.py

```diff
@@ -207,9 +207,12 @@
 ROOT_DIR = os.path.abspath(os.path.join(os.path.dirname(__file__), os.pardir))
 
 # Generate top-level
+
+allowed_top_level = ("exceptions.py",)
```

Comment:
What is this all about?

Reply:
ah, was wondering if this would come up. @potiuk suggested that I should add documentation to the concepts section, and it felt incomplete to me to add that without linking to the exceptions api docs. all modules at the top level (airflow/*.py) are excluded from the documentation and my link to the exceptions api doc could not be rendered and was breaking the docs build. adding this ... exception :) ... fixed that

i'm happy to take any direction whether that's leaving this as-is, removing the links to the exception api docs, or something in-between. just let me know what you'd prefer

Reply:
i can also just add a comment explaining why this exists

Reply:
Ah, the API is handled differently (we render it with an autodoc extension) - can you get away with just linking to

```
   :py.module:`airflow.exceptions`
```

where you want to and then this change might not be needed?

Reply:
Ah, the API is handled differently (we render it with an autodoc extension) - can you get away with just linking to

```
   :py.module:`airflow.exceptions`
```

where you want to and then this change might not be needed?

Reply:
This works ... 

```
:py:mod:`airflow.exceptions`
```

... but only if I keep my changes in conf.py 😦 Otherwise it "works" but does not create a link. I'm pretty sure the code I'm touching in conf.py is actively excluding that module from the autodocs.

Reply:
@mik-laj @potiuk Any thoughts on this specific change?

Reply:
Looking at it a bit more I think you are right though @jstern

Reply:
@ashb @potiuk @mik-laj - any thoughts about whether it's ok to keep this change to include the exceptions module in the docs?

---

On a related note, after updating this pr from master, I'm getting a new docs build failure:

```
/opt/airflow/docs/_api/airflow/providers/amazon/aws/hooks/datasync/index.rst:66: WARNING: more than one target found for cross-reference 'AirflowException': airflow.exceptions.AirflowException, airflow.providers.yandex.hooks.AirflowException
/opt/airflow/docs/_api/airflow/providers/amazon/aws/operators/datasync/index.rst:167: WARNING: more than one target found for cross-reference 'AirflowException': airflow.exceptions.AirflowException, airflow.providers.yandex.hooks.AirflowException
/opt/airflow/docs/_api/airflow/providers/amazon/aws/operators/datasync/index.rst:167: WARNING: more than one target found for cross-reference 'AirflowException': airflow.exceptions.AirflowException, airflow.providers.yandex.hooks.AirflowException
/opt/airflow/docs/_api/airflow/providers/amazon/aws/operators/datasync/index.rst:167: WARNING: more than one target found for cross-reference 'AirflowException': airflow.exceptions.AirflowException, airflow.providers.yandex.hooks.AirflowException
/opt/airflow/docs/_api/airflow/providers/amazon/aws/operators/datasync/index.rst:167: WARNING: more than one target found for cross-reference 'AirflowException': airflow.exceptions.AirflowException, airflow.providers.yandex.hooks.AirflowException
```

I can make this go away by removing my change that allows `airflow.exceptions` api docs to be built, but this does not seem like a satisfactory course of action, because the `airflow.providers.yandex.hooks.AirflowException` will remain in the docs and it shouldn't. It's there because somehow these lines in `airflow/providers/yandex/hooks/__init__.py`:

```
from airflow.exceptions import AirflowException
from airflow.hooks.base_hook import BaseHook
```

are causing autoapi to include `AirflowException` and `BaseHook` in the docs for `airflow.providers.yandex.hooks` as if those classes were defined in that module.

<img width="719" alt="Screen Shot 2020-02-27 at 4 25 31 PM" src="https://user-images.githubusercontent.com/617540/75493344-46b1d680-597f-11ea-9e2c-2b8d30d3c302.png">

In other words, autoapi is generating this in `docs/_api/airflow/providers/yandex/hooks/index.rst`:

```
Package Contents
----------------

.. py:exception:: AirflowException

   Bases: :class:`Exception`

   Base class for all Airflow's errors.
   Each custom exception should be derived from this class

   .. attribute:: status_code
      :annotation: = 500

.. py:class:: BaseHook
```

I don't know if this is a bug in autoapi or in how the airflow project configures it, and I don't know why the warnings are only about AirflowException and not BaseHook...